src/tsolvers/lrasolver/FarkasInterpolator.cc

```diff
@@ -630,10 +570,144 @@ PTRef LRA_Interpolator::getInterpolant(icolor_t color, PartitionManager &pmanage
     return itp;
 }
 
-bool LRA_Interpolator::isALocal(PTRef var, PartitionManager &pmanager) const {
+bool FarkasInterpolator::isALocal(PTRef var) const {
     return isAstrict(pmanager.getIPartitions(var), mask);
 }
 
-bool LRA_Interpolator::isBLocal(PTRef var, PartitionManager &pmanager) const {
+bool FarkasInterpolator::isBLocal(PTRef var) const {
     return isBstrict(pmanager.getIPartitions(var), mask);
 }
+
+PTRef FarkasInterpolator::getInterpolant(FarkasItpOptions const & options) {
```

Comment:
Is there a way of giving this function a bit more structure?  Separate the different ITP algorithm implementations to different functions, with those having a lot of code in common maybe further split into other functions?  It's a bit painful to read as it is now.